iceoryx_examples/complexdata/iox_publisher_complexdata.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "topic_data.hpp"
+
+#include "iceoryx_posh/popo/publisher.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "iceoryx_utils/posix_wrapper/signal_handler.hpp"
+
+std::atomic_bool killswitch{false};
```

Comment:
I would recommend using it shutdown or similar in all examples, but of course is inconsequential.

Reply:
To be consistent with most of the other examples, I would keep the name.

Reply:
I agree with Matthias. We should promote love, not killing ;)
This should be done in a separate PR, though ... maybe when geoffrey takes care of the syncing ;)